sqlair.go

```diff
@@ -165,17 +166,24 @@ func (db *DB) Query(ctx context.Context, s *Statement, inputArgs ...any) *Query
 		ctx = context.Background()
 	}
 
-	sqlstmt, err := db.prepareStmt(ctx, db.sqldb, s)
+	qe, tempStmt, err := s.pe.Query(inputArgs...)
```

Comment:
I am brainstorming about how to remove the extra return here and how to simply it a bit, especially thinking about later revamping the cache. What do you think of storing the array lengths in `Statement`? For now it will be the default length but in the future we can store the latest one in order to update the cache. That way below we can just check if the `inputArgs` lengths do not match and re-prepare and re-generate the SQL. The advantages are:
* Arguably it makes sense to store the length of the prepared statement in the struct that contains the rest of the information.
* We do not have to carry a temporal value in the loop inside `Query` or here.

Reply:
You are right about the concurrency, it makes more sense to later follow an approach like the one you described in the document.